### PR TITLE
Feature/employer messages

### LINKED ISSUE #54: Send message to applicants
- [x] A message icon on the Applicant component is shown when you are not logged in as that applicant
- [x] When that message icon is clicked, a form is rendered for sending a message
- [x] It won't send an empty message
- [ ] When that message is sent, a POST is made to the backend
- [ ] On successful post a message is shown

### PR DESCRIPTION
**Pull request by GreysonElkins**
#### Type of Change Made 
- [ ] bugfix 
- [x] new feature 
- [ ] breaking change 
- [ ] testing 
- [ ] formatting 
- [ ] styling
#### What's this PR do?
* Allows users to send messages to applicants
* Prevents applicants from messaging themselves 
* Handles several sad-path message actions
* Says 'You don't have any messages yet' if inbox is empty
* Only allows users to view their own inbox
#### Where should the reviewer start?
* MessageForm.tsx
#### How should this be manually tested?
Send some messages, go check the users inbox. (You may have to use the component inspector to fake a log in)
#### What are the relevant tickets?
close #54

### CODE REVIEW

File: src/components/Inbox/Inbox.tsx [hunk 1]
```diff
@@ -52,9 +54,22 @@ const Inbox: React.FC<RouteComponentProps> = (props) => {
...
   return (
-    <div className="message-container">    
-      {displayMessages()}
-    </div>
+    <AuthContext.Consumer>
+     {({ auth }) => {
+       const match:any = props.match.params
+       if(auth !== parseInt(match.id)) {
+        return (
+          <Redirect to={`/applicant/${match.id}`} />
```
Comment (ErinUntermeyer) [inline src/components/Inbox/Inbox.tsx:62]: Nice touch here!

File: src/components/MessageForm/MessageForm.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React, { useState, ChangeEvent } from 'react'
```
Comment (ErinUntermeyer) [inline src/components/MessageForm/MessageForm.tsx:1]: Wow, this is awesome. I can see it's much better this is its own component!

File: src/components/SearchResults/SearchResults.tsx [hunk 3]
```diff
@@ -13,12 +13,23 @@ const SearchResults: React.FC<RouteComponentProps> = (props) => {
   let query: unknown | any = props.location.state
 
   useEffect(() => {
+    const filterMatches = (data:SearchResponse) => {
```
Comment (ErinUntermeyer) [inline src/components/SearchResults/SearchResults.tsx:16]: Thank you for adding this filter method! Can't wait for that endpoint to be up and working so we can see some of the site in action. :)

**Review by ErinUntermeyer — APPROVED** [review]
Great job, thanks for getting this done today. I think that's MVP! Now just testing and some cleanup and we are good to go.

File: src/components/MessageForm/MessageForm.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  const [message, setMessage] = useState<EmployerMessage>({employer_name: '', employer_email: '', body: ''})
+  const [error, setError] = useState('')
+
+  const handleMessageChange = (event:ChangeEvent<HTMLTextAreaElement> ) => {
+    setMessage({...message, [event.target.id]: event.target.value})
+    setError('')
+  }
+
+  const postMessage = () => {
+    if (Object.keys(message).every((property:string) => {
```
Comment (amykarnaze) [inline src/components/MessageForm/MessageForm.tsx:20]: Great job accessing the messages here, Greyson.

**Comment by amykarnaze** [comment]
Great job implementing message creation by applicants and adding error handling on the inbox. You are so fast!

**Pull request merged**